app/src/main/java/com/example/drawingactivity/ForcableEditText.kt

```diff
@@ -0,0 +1,54 @@
+package com.example.drawingactivity
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.text.TextWatcher
+import android.util.AttributeSet
+import android.widget.EditText
+import com.example.drawingactivity.ColorConstants.rgbMax
+
+@SuppressLint("ClickableViewAccessibility")
+class ForcableEditText @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0
+) : EditText(context, attrs, defStyle) {
+
+    override fun addTextChangedListener(watcher: TextWatcher?) {
+        watcher?.let { if (!textListeners.contains(it)) textListeners.add(it) }
+        super.addTextChangedListener(watcher)
+    }
+
+    private val textListeners = mutableListOf<TextWatcher>()
+
+    private fun removeAllListeners() {
+        textListeners.forEach { removeTextChangedListener(it) }
+    }
+
+    private fun addBackAllListeners() {
+        textListeners.forEach { addTextChangedListener(it) }
+    }
+
+    fun forceText(text: String) {
+        if (!this.hasFocus()) {
+            removeAllListeners()
+            setText(text)
+            addBackAllListeners()
+        }
+    }
+
+    val colorInt: Int?
+        get() {
+            return this.text.toString().toIntOrNull()
+        }
+
+    val floatConvert: Float
+        get() {
+            return this.text.toString().toFloat()
+        }
+
+    val rgbIntConvert: Int
```

Comment:
What's this for? You already have an int function? If you want it as a color convert it to a color? I'm not sure what this is trying to do

Reply:
I misunderstood that the or null is actually converting it on success. Sloppy error

Reply:
This was being used in three places as a shorthand and didn't save any lines of code at the end of the day, it has been removed.